packages/code-studio/src/storage/grpc/GrpcFileStorageTable.ts

```diff
@@ -161,28 +174,59 @@ export class GrpcFileStorageTable implements FileStorageTable {
     };
   }
 
-  async refresh(): Promise<ViewportData<FileStorageItem>> {
-    log.debug2(this.root, 'refreshData');
+  doRefresh(callback: ViewportUpdateCallback<FileStorageItem>): void {
+    log.debug('doRefresh', this.root);
+
+    const viewportDataPromise = this.fetchData();
+
+    viewportDataPromise.then(callback).catch(e => {
+      if (!PromiseUtils.isCanceled(e)) {
+        log.error('Error fetching viewport data', e);
+      }
+    });
+  }
```

Comment:
I know why it was complaining about the unchained version of this. When calling `viewportDataPromise.then(callback)`, that creates a new promise (each `.then` or `.catch` creates a new promise), and that new promise did not have a catch on it so was uncaught.

A better way to handle this (than silently eating the error) is to actually just return the promise, rather than resolve it in the `doRefresh` method. Then the `refresh()` method can still set the `viewportUpdatePromise` correctly, and a call to `getViewportData()` will reject correctly if the operation was cancelled.

```suggestion
  doRefresh(
    callback: (fetchPromise: Promise<ViewportData<FileStorageItem>>) => void
  ): void {
    log.debug2('doRefresh', this.root);
    callback(this.fetchData());
  }
```